mach crashes trying to send notifications on Windows Server 2019

As of 00f267e2893a6dbca2c54f9669e2c431b59ecd19:

```
Error running mach:

    ['build', '--locked', '--release']

The error occurred in code that was called by the mach command. This is either
a bug in the called code itself or in the way that mach is calling it.
You can invoke |./mach busted| to check if this issue is already on file. If it
isn't, please use |./mach busted file| to report it. If |./mach busted| is
misbehaving, you can also inspect the dependencies of bug 1543241.

If filing a bug, please include the full output of mach, including this error
message.

The details of the failure are as follows:

notifypy.exceptions.UnsupportedPlatform: This version of Windows (2019Server) is not supported. is not supported.

  File "C:\a\servo\servo\python\servo\command_base.py", line 734, in configuration_decorator
    return original_function(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\a\servo\servo\python\servo\build_commands.py", line 153, in build
    self.notify("Servo build", build_message)
  File "C:\a\servo\servo\python\servo\build_commands.py", line 260, in notify
    notification = notifypy.Notify(use_custom_notifier=notifier)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\a\servo\servo\python\_venv3.12\Lib\site-packages\notifypy\notify.py", line 69, in __init__
    self._notifier_detect = self._selected_notification_system()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\a\servo\servo\python\_venv3.12\Lib\site-packages\notifypy\notify.py", line 123, in _selected_notification_system
    raise UnsupportedPlatform(
Error: Process completed with exit code 1.
```

--- a/python/servo/build_commands.py
+++ b/python/servo/build_commands.py
@@ -256,12 +256,15 @@ class MachCommands(CommandBase):
                 print("[Warning] Could not generate notification: "
                       f"Could not run '{notify_command}'.", file=sys.stderr)
         else:
-            notifier = LinuxNotifier if sys.platform.startswith("linux") else None
-            notification = notifypy.Notify(use_custom_notifier=notifier)
-            notification.title = title
-            notification.message = message
-            notification.icon = path.join(self.get_top_dir(), "resources", "servo_64.png")
-            notification.send(block=False)
+            try:
+                notifier = LinuxNotifier if sys.platform.startswith("linux") else None
+                notification = notifypy.Notify(use_custom_notifier=notifier)
+                notification.title = title
+                notification.message = message
+                notification.icon = path.join(self.get_top_dir(), "resources", "servo_64.png")
+                notification.send(block=False)
+            except notifypy.exceptions.UnsupportedPlatform as e:
+                print(f"[Warning] Could not generate notification: {e}", file=sys.stderr)
 
 
 def otool(s):

